src/challenge/Challenge.tsx

```diff
@@ -230,7 +237,20 @@ class Challenge extends React.Component<ChallengeProps, ChallengeState> {
 
   handleDownload = () => {
     this.editorRef.current?.download();
-  };  
+  };
+
+  handleInputModeChange = (inputMode:boolean) => {
+    this.setState({isFixedInput: inputMode})
+  }
+
+  getFixedInputValue = () => {
+    return this.fixedUserInput.shift();
```

Comment:
this could go in the controller.

Reply:
yes very helpful: things are beginning to become clearer now - I had thought perhaps all react apps required everything to be either in props or state and then using refs etc but now it all makes more sense.

Reply:
moved